### PR TITLE
performance benchmarking Jenkinsfile step1

### PR DESCRIPTION
**Pull request by whchung**
- suppress result tensor print out logic from host logic populated by `mlir-miopen-driver`.
- add a new Jenkinsfile to test just 1 config and dump the perf result using `rocprof`.

will collect more configs after hooking up with MIOpen perfdb using sqlite.

### CODE REVIEW

**Comment by whchung** [comment]
also need to figure out how to add a gfx908 node into the federation of CI hosts..

File: mlir/utils/jenkins/Jenkinsfile.perf [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                        -DLLVM_TARGETS_TO_BUILD="X86;NVPTX;AMDGPU" \
+                        -DCMAKE_BUILD_TYPE=Release \
+                        -DLLVM_ENABLE_ASSERTIONS=ON \
+                        -DBUILD_SHARED_LIBS=ON \
+                        -DLLVM_BUILD_LLVM_DYLIB=ON \
+                        -DMLIR_ROCM_RUNNER_ENABLED=1 \
+                        -DMLIR_MIOPEN_DRIVER_ENABLED=1
+
+                    # build LLVM / MLR and run tests
+                    cmake --build . --target check-mlir
```
Comment (jerryyin) [inline mlir/utils/jenkins/Jenkinsfile.perf:45]: I feel like we can skip the sanity check and make this performance one completely orthogonal to it. Though I don't feel strong about it and leaves the decision to you.

File: mlir/utils/jenkins/Jenkinsfile.perf [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+                    # run 1 perf benchmark config. more to come.
+                    ./bin/mlir-miopen-driver -p -ph -c | rocprof --hip-trace ./bin/mlir-rocm-runner --shared-libs=./lib/librocm-runtime-wrappers.so,./lib/libmlir_runner_utils.so --entry-point-result=void
+
+                    # dump perf figures.
+                    cat results.stats.csv
```
Comment (jerryyin) [inline mlir/utils/jenkins/Jenkinsfile.perf:51]: Could this stay in a unit test by itself? It feels strange that we need to update the Jenkins file to add performance related tests.

File: mlir/utils/jenkins/Jenkinsfile.perf (hunk 1 above)
Comment (whchung) [reply mlir/utils/jenkins/Jenkinsfile.perf:45]: this step is a sanity check, plus also making sure all dependent tools are built.

File: mlir/utils/jenkins/Jenkinsfile.perf (hunk 2 above)
Comment (whchung) [reply mlir/utils/jenkins/Jenkinsfile.perf:51]: this is not a unit test, but a performance benchmarking suite.
the idea is to drive it via a Jenkinsfile, and produce benchmarking results and dump them in the console.

Comment (jerryyin) [reply mlir/utils/jenkins/Jenkinsfile.perf:51]: We can worry about this later when there's a variety of performance bench-marking comes to play. By then we'd probably needs a dedicated thing to process it.

**Review by jerryyin — APPROVED** [review]

**Pull request merged**